src/v0/functions.ts

```diff
@@ -0,0 +1,250 @@
+import qs from 'qs'
+import { Client } from '../client'
+import { BaseError } from '../errors'
+import { UriHelper } from '../uri-helper'
+import { ThBaseHandler } from '../base'
+
+export interface FunctionsOptions {
+  user?: string
+  base?: string
+}
+
+export interface FunctionsQuery {
+  limit?: number
+  uri?: string
+  query?: {
+    deleted?: boolean
+    active?: boolean
+  }
+}
+
+export interface FunctionsResponse {
+  data: object[]
+  metadata: object
+  next?: () => Promise<FunctionsResponse>
+}
+
+export interface FunctionResponse {
+  data: Function
+  metadata?: {
+    count?: number
+    patch?: any
+  }
+  msg?: string
+}
+
+export type FunctionRuntime = 'pos' | 'nodejs8x' | 'python27'
+export type FunctionType = 'local' | 'http' | 'pubsub'
+export type FunctionConfigurationClass = 'instant_checkout' | 'add_to_cart'
+export type FunctionConfigurationResourceType = 'product' | 'discount' | 'voucher_action' | 'customer'
+
+export interface FunctionConfigurationResource {
+  type?: FunctionConfigurationResourceType
+  object_id?: string
+}
+
+export interface FunctionConfigurationImages {
+  original?: string
+  '1x'?: string
+  avatar?: string
+}
+
+export interface FunctionConfiguration {
+  class: FunctionConfigurationClass
+  description?: string
+  resources?: FunctionConfigurationResource[]
+  images?: FunctionConfigurationImages
+  // TODO: not sure what to do with instantCheckoutPayload
+}
+
+export interface Function {
+  name?: string
+  runtime?: FunctionRuntime
+  type?: FunctionType
+  function?: string
+  topic?: string
+  handler?: string
+  deps?: string
+  active?: boolean
+  deleted?: boolean
+  configuration?: FunctionConfiguration
+}
+
+export class Functions extends ThBaseHandler {
+  public static baseEndpoint = '/api/v0/functions'
+  endpoint: string
+  http: Client
+  public options: FunctionsOptions
+  public uriHelper: UriHelper
+
+  constructor(options: FunctionsOptions, http: Client) {
+    super(http, { endpoint: Functions.baseEndpoint, base: options.base || 'https://api.tillhub.com' })
+    this.options = options
+    this.http = http
+
+    this.endpoint = Functions.baseEndpoint
+    this.options.base = this.options.base || 'https://api.tillhub.com'
+    this.uriHelper = new UriHelper(this.endpoint, this.options)
+  }
+
+  getAll(queryOrOptions?: FunctionsQuery | undefined): Promise<FunctionsResponse> {
+    return new Promise(async (resolve, reject) => {
+      let next
+
+      try {
+        let uri
+        if (queryOrOptions && queryOrOptions.uri) {
+          uri = queryOrOptions.uri
+        } else {
+          let queryString = ''
+          if (queryOrOptions && (queryOrOptions.query || queryOrOptions.limit)) {
+            queryString = qs.stringify({ limit: queryOrOptions.limit, ...queryOrOptions.query })
+          }
+
+          uri = `${this.options.base}${this.endpoint}/${this.options.user}${
+            queryString ? `?${queryString}` : ''
+            }`
```

Comment:
Is there a specific reason you're not using the `UriHelper` here?

Reply:
I copy-pasted the devices endpoint and replaced device with function. Should I change it to use UriHelper?

Reply:
Yes, please replace. It's legacy and we don't want to do stuff this way anymore.